### ISSUE TITLE
FastCreateList fails on collections with no constructor with a capacity parameter

### ISSUE THREAD

**Issue by scpurcell** [mgholam/fastJSON#87]
I noticed after updating to version 2.1.36 that a FillObject call which worked fine in version 2.1.34 now failed.  This parameter is a navigation link for Entity Framework, this is its typical declaration:

        public virtual ObservableCollection<Item> Sources { get; set; }

The exception:

```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Exception: Failed to fast create instance for type 'System.Collections.ObjectModel.ObservableCollection`1[[Project.Models.Item, Project, Version=1.0.6801.25521, Culture=neutral, PublicKeyToken=null]]' from assembly 'System.Collections.ObjectModel.ObservableCollection`1[[Project.Models.Item, Project, Version=1.0.6801.25521, Culture=neutral, PublicKeyToken=null]], System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' ---> System.ArgumentNullException: Value cannot be null.
Parameter name: con
   at System.Reflection.Emit.DynamicILGenerator.Emit(OpCode opcode, ConstructorInfo con)
   at fastJSON.Reflection.FastCreateList(Type objtype, Int32 capacity)
```

**Comment by mgholam** [comment]
Ah! to do with the list optimization... 

Do you have a test case for this (most of the .net collections have a capacity ctor)?

**Comment by scpurcell** [comment]
Here's a quick test which demonstrates the exception:

```
using System;
using System.Collections.ObjectModel;

namespace ConsoleApp1
{
    public class Item
    {
        public int Id { get; set; }
        public string Data { get; set; }
    }

    public class TestObject
    {
        public int Id { get; set; }
        public string Stuff { get; set; }
        public virtual ObservableCollection<Item> Items { get; set; }
    }

    class Program
    {
        static void Main(string[] args)
        {
            TestObject testObject = new TestObject
            {
                Id = 1,
                Stuff = "test",
                Items = new ObservableCollection<Item>()
            };

            testObject.Items.Add(new Item { Id = 1, Data = "Item 1" });
            testObject.Items.Add(new Item { Id = 2, Data = "Item 2" });

            string jsonData = fastJSON.JSON.ToNiceJSON(testObject);
            Console.WriteLine(jsonData);

            TestObject copyObject = new TestObject();
            fastJSON.JSON.FillObject(copyObject, jsonData);
        }
    }
}

**Comment by mgholam** [comment]
Fixed in v2.1.37
